### ISSUE TITLE
[Feature Request] Add setting to automatically group windows of a certain application

### ISSUE THREAD

**Issue by Hobart2967** [Windows-Apps-Hub/UnitedSets#52]
As a VSCode User, I would love to have all my VSCode windows being grouped on opening, to not have the hassle to open a new window, open my united set, add a tab there then head back to my window, select it, then pick up the other one and add it as a tab as well.

**Comment by mitchcapper** [comment]
This could be rule based grouping,  or (potentially more useful) is simply child grouping.  The option to have all children of an application automatically appear in the same US.

**Comment by GetGet99** [comment]
Hmm... I'm not 100% sure if I understand this. My understanding is that when a new window appears from some certain application lists, you'd like to automatically add it into United Sets. Is this what you mean? If not, can you clarify?

**Comment by Hobart2967** [comment]
Yes exactly. To me, it's not neccessarily required to have some customizable application groups, though it's gonna be nice.

I'd expect this to work like it does on MacOS, where upon opening a new window, it is automatically added to the existing group.

See following video (Strg+Shift+N opens a new window, which is the hot-key-stroke I pressed).

![Image](https://github.com/user-attachments/assets/07013a33-af8a-4f17-ab1b-006b5883ddcf)
